appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jmac/ConfigDomainParser.java

```diff
@@ -92,34 +106,24 @@ private void processServerConfig(SecurityService service, Map newConfig) throws
 
                 if (provList != null) {
 
-                    for (ProviderConfig provider : provList) {
+                    Iterator<ProviderConfig> pit = provList.iterator();
```

Comment:
Looks like a revert of recent improvements in master branch. For loop is better than Iterator.